graphql/resolvers/directives.js

```diff
@@ -0,0 +1,73 @@
+import fs from 'fs';
+import path from 'path';
+import { forEachField } from 'graphql-tools';
+import { getArgumentValues } from 'graphql/execution/values';
+import jwt from 'jsonwebtoken';
+import debug from 'debug';
+
+import { AuthorizationError } from '../errors';
+
+const log = debug('fcc:resolvers:directives');
+const certPath = process.cwd() + '/public.pem';
+const cert = fs.readFileSync(certPath);
+
+function isAuth(ctx) {
+  const token = ctx.headers.authorization;
+  if (!token) {
+    throw new AuthorizationError({
+      message: 'You must supply a JSON Web Token for authorization!'
+    });
+  }
+  try {
+    return jwt.verify(token.replace('Bearer ', ''), cert);
+  } catch (err) {
+    return false;
+  }
+}
+
+const directiveResolvers = {
+  isAuthenticatedOnField: (result, source, args, ctx) =>
+    isAuth(ctx) ? result : null,
+  isAuthenticatedOnQuery: (result, source, args, ctx) => {
+    const hasAuth = isAuth(ctx);
+    if (hasAuth) {
+      return result;
+    }
+    throw new AuthorizationError({
+      message: `You are not authorized. The required permission was not found.`
+    });
+  }
+};
+
+// Credit: agonbina https://github.com/apollographql/graphql-tools/issues/212
```

Comment:
Is this still needed after https://github.com/apollographql/graphql-tools/pull/518 was merged?

Reply:
I didn't even realise this was a thing inside `graphql-tools`, I will update the PR to let `graphql-tools` worry about the implementation